controllers/spacebindingcleanup/spacebinding_cleanup_controller.go

```diff
@@ -73,9 +80,18 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 	if err := r.Client.Get(context.TODO(), spaceName, space); err != nil {
 		if errors.IsNotFound(err) {
 			logger.Info("the Space was not found", "Space", spaceBinding.Spec.Space)
-			return ctrl.Result{}, r.deleteSpaceBinding(logger, spaceBinding)
-
+			if requeueAfter, err := r.deleteSpaceBinding(logger, spaceBinding); err != nil {
+				return ctrl.Result{}, err
+			} else if requeueAfter > 0 {
+				return ctrl.Result{
+					Requeue:      true,
+					RequeueAfter: requeueAfter,
+				}, nil
+			}
+			// spacebinding deleted and no reconcile needed
+			return ctrl.Result{}, nil
```

Comment:
from the `Result` documentation:
```
// RequeueAfter if greater than 0, tells the Controller to requeue the reconcile key after the Duration.
// Implies that Requeue is true, there is no need to set Requeue to true at the same time as RequeueAfter.
RequeueAfter time.Duration
```
you could simplify it:
```suggestion
			requeueAfter, err := r.deleteSpaceBinding(logger, spaceBinding)
			return ctrl.Result{
				RequeueAfter: requeueAfter,
			}, err
```

Reply:
Ha nice, thanks for pointing that out!